src/lib/viewers/doc/DocFirstPreloader.js

```diff
@@ -436,6 +437,73 @@ class DocFirstPreloader extends EventEmitter {
         });
     }
 
+    readEXIFNew(imageBlob, imageEl) {
+        return new Promise((resolve, reject) => {
+            try {
+                let tags = {};
+                const reader = new FileReader();
+                reader.onload = () => {
+                    const arrayBuffer = reader.result;
+                    /* global ExifReader */
+                    tags = ExifReader.load(arrayBuffer);
+
+                    const userComment = tags.UserComment.description || tags.UserComment.value;
+                    const match = EXIF_COMMENT_REGEX.exec(userComment);
+
+                    // There should be 3 pieces of metadata: PDF width, PDF height, and num pages
+                    if (!match || match.length !== 4) {
+                        reject(new Error('No valid EXIF data found'));
+                        return;
+                    }
+
+                    // Convert PDF Units to CSS Pixels
+                    let pdfWidth = parseInt(match[1], 10) * PDFJS_CSS_UNITS;
+                    let pdfHeight = parseInt(match[2], 10) * PDFJS_CSS_UNITS;
+                    const numPages = parseInt(match[3], 10);
+
+                    // Validate number of pages
+                    if (numPages <= 0) {
+                        reject(new Error('EXIF num pages data is invalid'));
```

Comment:
A lot of thrown errors in this file, do we have tests for all of these scenarios?